src/Database/Type/DateTimeType.php

```diff
@@ -212,22 +250,7 @@ public function manyToPHP(array $values, array $fields, DriverInterface $driver)
                 continue;
             }
 
-            if (strpos($values[$field], '0000-00-00') === 0) {
-                $values[$field] = null;
-                continue;
-            }
-
-            $instance = clone $this->_datetimeInstance;
-            $instance = $instance->modify($values[$field]);
-            if ($instance->getTimezone()->getName() !== date_default_timezone_get()) {
-                $instance = $instance->setTimezone(new DateTimeZone(date_default_timezone_get()));
-            }
-
-            if ($this->setToDateStart) {
-                $instance = $instance->setTime(0, 0, 0);
-            }
-
-            $values[$field] = $instance;
+            $values[$field] = $this->toPHP($values[$field], $driver);
```

Comment:
I think the goal with the duplication here was to avoid the overhead of additional method calls. While small these add up in aggregate and the `manyToPHP()` methods are intended to be an optimization.

Reply:
I can re-build it.  I changed it to see what the team thought.